module/VuFindSearch/src/VuFindSearch/Backend/EDS/Base.php

```diff
@@ -494,6 +534,14 @@ protected function call(
         if ($this->isGuest && !empty($this->apiKeyGuest)) {
             $headers['x-api-key'] = $this->apiKeyGuest;
         }
+        if ($this->sendUserIp) {
+            $headers['x-eis-enduser-ip-address'] = $this->ipToReport ?? '-';
+            $headers['x-eis-enduser-user-agent'] = $this->userAgent ?? 'No user agent';
+            $headers['x-eis-vendor'] = 'VuFind';
+            if (!empty($this->reportVendorVersion)) {
+                $headers['x-eis-vendor-version'] = $this->reportVendorVersion;
+            }
+        }
```

Comment:
It might make sense to refactor all of the code that builds the `$headers` array to a separate `protected function getRequestHeaders(array $headerParams = []): array`. That way, if somebody wants to customize something like `x-eis-vendor` they could easily do so through a child class without having to rewrite the whole call() function. It makes things more customizable without requiring you to worry about extra config settings, etc.

Also, a related note: the function documentation here suggests that `$headerParams` must be an array, but the code supports it being null. I'm not sure where this inconsistency came from, but it's something we probably ought to address at some point. Not sure if it's worth doing as part of this work, but for now, I'd suggest only accepting an array in `getRequestHeaders` and then calling it here with `$this->getRequestHeaders($headerParams ?? [])`. That way, you can eliminate the null check (since iterating an empty array seems simpler than adding a null check) but still maintain backward compatibility with the existing behavior.

Reply:
not sure how to do it yet, but I like that idea :) Thank you for your patience with my Pull Requests and sorry for ruining your vacation...

Reply:
If you'd like me to open a separate PR to do the header refactoring so that you can then merge the changes into here, I can do that. And aren't you supposed to be on vacation too? ;-)

Reply:
Half the fun to learn something new :) You might end up having to help me with pointers, but if you are ok with it, I'd like to try it.

You might have noticed that I am posting from my private account, because I am indeed on vacation. But jet lag is terrible and this gives me something useful to do before the sun comes up, and even at work this is not rally my "job", but I enjoy doing it - and the community...

Reply:
Yes, please feel free, and I'm happy to help if you have questions along the way!